src/movies.js

```diff
@@ -1,16 +1,108 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
+function getAllDirectors(movies) {
+    let answer = movies.map(function(movie) {
+        return movie.director   
+    });
+    return answer
+}
+
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(movies) {
+    
+    let result = movies.filter(function(movie) {
+        // Condition: Only drama movies & from Steven Spielberg
+        if (movies.director == "Steven Spielberg" && movies.genre.includes("Drama")) {
```

Comment:
instead of movies.director and movies.genre it should be movie.director and movie.genre . Remember movies is the original array being passed from the main function parameter, movie is each movie element being passed from the filter parameter.